packages/transport/src/modules/messages/MessageController.ts

```diff
@@ -62,6 +62,25 @@ export class MessageController extends TransportController {
         });
     }
 
+    public async deleteRecipientInMessage(messageId: CoreId, relationship: Relationship): Promise<void> {
+        const messageDoc = await this.messages.read(messageId.toString());
+        const message = Message.from(messageDoc);
+
+        message.relationshipIds = message.relationshipIds.filter((id) => !id.equals(relationship.id));
+        if (message.relationshipIds.length === 0) {
+            await this.messages.delete(message);
+            return;
+        }
+        await this.updateCacheOfMessage(message);
+
+        await this.messages.update(messageDoc, message);
+    }
+
+    public async decomposeMessagesOfRelationship(relationship: Relationship): Promise<void> {
```

Comment:
We _decompose_ a relationship but not messages. Why not something like `deletePeerOfRelationshipFrom(ReceipientsOf)Messages`?